devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -56,23 +64,88 @@ metadata {
         fingerprint mfr: "0312", prod: "C000", model: "C009", deviceJoinName: "Minoston Outlet" // old MP21Z
         fingerprint mfr: "0312", prod: "FF00", model: "FF0C", deviceJoinName: "Minoston Outlet" //MP21Z Minoston Mini Smart Plug
         fingerprint mfr: "0312", prod: "AC01", model: "4001", deviceJoinName: "New One Outlet"  // N4001 New One  Mini Smart Plug
+        fingerprint mfr: "0312", prod: "EE00", model: "EE01", deviceJoinName: "Minoston Switch" //MS10ZS Minoston Smart Switch
+        fingerprint mfr: "0312", prod: "EE00", model: "EE03", deviceJoinName: "Minoston Switch" //MS12ZS Minoston Smart on/off Toggle Switch
+        fingerprint mfr: "0312", prod: "A000", model: "A005", deviceJoinName: "Evalogik Switch" //ZW30
+        fingerprint mfr: "0312", prod: "BB00", model: "BB01", deviceJoinName: "Evalogik Switch" //ZW30S Evalogik Smart on/off Switch
+        fingerprint mfr: "0312", prod: "BB00", model: "BB03", deviceJoinName: "Evalogik Switch" //ZW30TS Evalogik Smart on/off Toggle Switch
     }
 
     preferences {
-        configParams.each {
-            if (it.name) {
-                if (it.range) {
-                    input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
-                } else {
-                    input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
-                }
+        getConfigParamInput(ledModeParam)
+        getConfigParamInput(autoOffIntervalParam)
+        getConfigParamInput(autoOnIntervalParam)
+        getConfigParamInput(powerFailureRecoveryParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Paddle Control' and 'createButton' are only valid for the devices with product number of MS10ZS, MS12ZS, ZW30, ZW30S, ZW30TS(one of them)",
+                required: false
+        getConfigParamInput(paddleControlParam)
+        input(type: "enum", name: "createButton", required: false, title: "Create Button for Paddles?", options: ["No", "Yes"], defaultValue:"Yes")
```

Comment:
Lines 75-78 and 83,84 are not new device parameters. Lines 79-82 are added to remind people that the next two device parameters are only valid to parts of products (because they have to be changed to a fixed display ).
The contents of lines 75-78 and 83 were previously displayed like this:
             configParams.each {
            if (it.name) {
                if (it.range) {
                    input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
                } else {
                    input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
                }
            }
        }
In order to add a reminder in the middle, I modified the display of the traversal mode to display directly.